src/coreclr/vm/threadstatics.cpp

```diff
@@ -799,6 +803,23 @@ bool CanJITOptimizeTLSAccess()
     // Optimization is disabled for FreeBSD/arm64
 #elif defined(FEATURE_INTERPRETER)
     // Optimization is disabled when interpreter may be used
+#elif !defined(TARGET_OSX) && defined(TARGET_UNIX) && defined(TARGET_ARM64)
+    // Optimization is enabled for linux/arm64 only for static resolver.
+    // For static resolver, the TP offset is same for all threads.
+    // For dynamic resolver, TP offset returned is that of a JIT thread and
+    // will be different for the executing thread.
+    uint32_t* resolverAddress = reinterpret_cast<uint32_t*>(GetTLSResolverAddress());
+    if (
+        // nop or hint 32
+        ((resolverAddress[0] == 0xd503201f) || (resolverAddress[0] == 0xd503241f)) &&
```

Comment:
Does this work correctly for musl glibc?

Reply:
Currently, we do not do this optimization for musl as you can see few lines above - https://github.com/dotnet/runtime/pull/104408/files/6ecf11fec6c07aefc61f3647a3b3e358e35b6008#diff-838c5a095d88dfabb928a305f3d4ec888cbb0241f0c0d5cf1bb802c706bd1edbR800-R801

Reply:
Is that because we end up with dynamic resolver on musl? It would be good to find out if this fix covers the musl too (and that ifdef above can be deleted) or if there is something else going on with musl.

Reply:
> Is that because we end up with dynamic resolver on musl?

We originally disabled it, probably because of same reason, but do you think that enabling for musl/arm64 can be done as a separate change?

Reply:
> We should understand what's going on with musl before backporting to make sure we are not missing anything.

Just to make sure I understand, if it turns out to be same issue for musl as well, we do not want to enable musl/arm64 for 8.0, right?

Reply:
Right, the servicing fix should be the minimal change.

Reply:
Anything else needed for this PR then?